File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/Merchandising/CategoriaMerchandisingForm.ts
import { StringEditor, LookupEditor, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CategoriaMerchandisingForm {
    LocalSap: StringEditor;
    TipoSucursal: LookupEditor;
    RutaDli: StringEditor;
    TipoSenalizacion: LookupEditor;
    ScreenDisplay: LookupEditor;
    TramosLisos: LookupEditor;
    Checkout: LookupEditor;
    M2Calc: LookupEditor;
    MedidaCabecera: LookupEditor;
    EndCap: LookupEditor;
    MedidaGrafico: LookupEditor;
    BusStop: LookupEditor;
    Aretes: LookupEditor;
    CabecerasInstitucionales: LookupEditor;
    ExhibidorRetail: LookupEditor;
    ExhibidorGloblaBrands: LookupEditor;
    ExhibidorWellBeginnings: LookupEditor;
    ExhibidorInstitucional: LookupEditor;
    ExhibidorMascarillas: LookupEditor;
    ExhibidorGenerico: LookupEditor;
    TamanoMiniheders: LookupEditor;
    TamanoHeader: LookupEditor;
    MedidasPecheras: LookupEditor;
    MedidaCopete: IntegerEditor;
    MedidasCanceleria: LookupEditor;
    TramosFarma: LookupEditor;
    PortaposterCanceleria: LookupEditor;
}

export class CategoriaMerchandisingForm extends PrefixedContext {
    static formKey = 'Merchandising.CategoriaMerchandising';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoriaMerchandisingForm.init)  {
            CategoriaMerchandisingForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;
            var w2 = IntegerEditor;

            initFormType(CategoriaMerchandisingForm, [
                'LocalSap', w0,
                'TipoSucursal', w1,
                'RutaDli', w0,
                'TipoSenalizacion', w1,
                'ScreenDisplay', w1,
                'TramosLisos', w1,
                'Checkout', w1,
                'M2Calc', w1,
                'MedidaCabecera', w1,
                'EndCap', w1,
                'MedidaGrafico', w1,
                'BusStop', w1,
                'Aretes', w1,
                'CabecerasInstitucionales', w1,
                'ExhibidorRetail', w1,
                'ExhibidorGloblaBrands', w1,
                'ExhibidorWellBeginnings', w1,
                'ExhibidorInstitucional', w1,
                'ExhibidorMascarillas', w1,
                'ExhibidorGenerico', w1,
                'TamanoMiniheders', w1,
                'TamanoHeader', w1,
                'MedidasPecheras', w1,
                'MedidaCopete', w2,
                'MedidasCanceleria', w1,
                'TramosFarma', w1,
                'PortaposterCanceleria', w1
            ]);
        }
    }
}

File: MasterDirectory/MasterDirectory.Web/Modules/Merchandising/CategoriaMerchandising/CategoriaMerchandisingForm.cs
using Serenity.ComponentModel;
using System;

namespace MasterDirectory.Merchandising.Forms;

[FormScript("Merchandising.CategoriaMerchandising")]
[BasedOnRow(typeof(CategoriaMerchandisingRow), CheckNames = true)]
public class CategoriaMerchandisingForm
{
    [Tab("General")]
    //[LookupEditor("CatMerchandising")]
    public string LocalSap { get; set; }
    public int TipoSucursal { get; set; }
    public string RutaDli { get; set; }
    public int TipoSenalizacion { get; set; }
    public int ScreenDisplay { get; set; }
    public int TramosLisos { get; set; }
    public int Checkout { get; set; }
    public string RampaDiscapa { get; set; }

    [Tab("Medidas")]

    public int M2Calc { get; set; }
    public int MedidaCabecera { get; set; }
    public int EndCap { get; set; }
    public int MedidaGrafico { get; set; }
    public int BusStop { get; set; }
    public int Aretes { get; set; }
    public int CabecerasInstitucionales { get; set; }

    [Tab("Exhibidores")]

    public int ExhibidorRetail { get; set; }
    public int ExhibidorGloblaBrands { get; set; }
    public int ExhibidorWellBeginnings { get; set; }
    public int ExhibidorInstitucional { get; set; }
    public int ExhibidorMascarillas { get; set; }
    public int ExhibidorGenerico { get; set; }

    [Tab("Tamaños")]

    public int TamanoMiniheders { get; set; }
    public int TamanoHeader { get; set; }
    public int MedidasPecheras { get; set; }
    public int MedidaCopete { get; set; }
    public string MedidasCanceleria { get; set; }
    public int TramosFarma { get; set; }
    public int PortaposterCanceleria { get; set; }
}